cookbook/networking/background-parsing.md

```diff
@@ -0,0 +1,240 @@
+  
+## 1. Add the `http` and `isolate` packages
+
+First, we'll want to add the [`http`](https://pub.dartlang.org/packages/http) 
+and [`isolate`](https://pub.dartlang.org/packages/isolate) packages to our 
+project. The `http` package makes it easier to perform network requests, and 
+the `isolate` package makes it easier to work with `Isolates`!
+
+```yaml
+dependencies:
+  http: <latest_version>
+  isolate: <latest_version>
+```
+  
+## 2. Make a network request
+
+In this example, we'll fetch a json large document that contains a list of 5000 
+photo objects from the [JSONPlaceholder REST API](https://jsonplaceholder.typicode.com/) 
+using the [`http.get`](https://docs.flutter.io/flutter/package-http_http/package-http_http-library.html) 
+method. 
+
+```dart
+Future<http.Response> fetchPhotos(http.Client client) async {
+  return client.get('https://jsonplaceholder.typicode.com/photos');
+}
+```
+
+Note: We're providing an `http.Client` to the function in this example. This
+will make the function easier to test and use in different environments!
+
+## 3. Parse and Convert the json into a List of Photos
+
+Next, following the guidance from the [Fetch data from the internet](/cookbook/networking/fetch-data/)
+recipe, we'll want to convert our `http.Response` into a list of Dart objects.
+This will make the data easier to work with in the future.
+
+### Create a `Photo` class
+
+First, we'll need to create a `Photo` class that contains data about a photo. 
+We will also include a `fromJson` factory to make it easy to create a `Photo` 
+starting with a json object.
+
+```dart
+class Photo {
+  final int id;
+  final String title;
+  final String thumbnailUrl;
+
+  Photo({this.id, this.title, this.thumbnailUrl});
+
+  factory Photo.fromJson(Map<String, dynamic> json) {
+    return new Photo(
+      id: json['id'] as int,
+      title: json['title'] as String,
+      thumbnailUrl: json['thumbnailUrl'] as String,
+    );
+  }
+}
+```
+
+### Convert the `http.Response` into a `List<Photo>`
+
+Now, we'll update the `fetchPhotos` function to return a `Future<List<Photo>>`. 
+To do so, we'll need to:
+
+  1. Parse the response body using the `dart:convert` package
+  2. Convert the parsed response into a `List<Photo>`
+
+```dart
+Future<List<Photo>> fetchPhotos(http.Client client) async {
+  final response =
+      await client.get('https://jsonplaceholder.typicode.com/photos');
+  final parsed = json.decode(response.body);
+
+  return parsed.map((json) => new Photo.fromJson(json)).toList();
+}
+```
+
+## 4. Move this work to a separate isolate
+
+If you run the `fetchPhotos` function on a slower phone, you may notice the app 
+freezes for a brief moment as it parses and converts the json. This is jank, 
+and we want to be rid of it!
+
+So how can we do that? By moving the parsing and conversion to a background
+isolate. To do so, we'll need to:
+
+  1. Create an `IsolateRunner`. This will allow us to run the `fetchPhotos`
```

Comment:
you may wish to consider the compute() method for this

Reply:
Ah, perfect, that's even simpler! Thanks :)